src/react-render-perf/lesson-02/README.md

```diff
@@ -1,5 +1,142 @@
 [Home](../README.md)
 
-# 02 - Prevent Context From Rerendering
+# 02 - Prevent Context From Re-rendering
 
-TODO
+React Context is useful for sharing data with its descendent components. When
+changes are made to the context, this causes the context and all of its
+descendents to be re-rendered. If the number of descendents is large, this can
+result in poor rendering performance.
+
+In order to avoid re-rendering all of the descendents we'd like only those
+components using the context to re-render.  Ideally, the should only re-render
+if the data that they're making use of changes.
+
+In the following example, everytime we update the `value` in the context, we
+re-render both child components of the context even though only one of them
+needs to be update.
+
+```tsx
+import {createContext, useContext, useState} from "react";
+
+type FooBar = {
+    foo: number;
+    bar: number;
+};
+
+const FooBarContext = createContext<FooBar>({foo: 0, bar: 0});
+
+const Foo = () => {
+    const {foo} = useContext(FooBarContext);
+    return <h1>foo = {foo}</h1>;
+};
+
+const Bar = () => {
+    const {bar} = useContext(FooBarContext);
+    return <h1>bar = {bar}</h1>;
+};
```

Comment:
hmmm this example is confusing to me, because in fact both child components are using the context. I think the point would be clearer if `Bar` didn't use the context at all, but did some expensive render-time computation (and had a `Date.now()` rendered for good measure).

Reply:
This example never gets rendered so I'm not sure how useful the `Date.now()` would be.

Reply:
oh yeah, if it's not actually run then the date.now isn't needed

Reply:
Also, the expensive render-time computation case is covered by Lesson #1.  This particular lesson is showing how using Context can often result in unnecessary renders.